access/common/config.go

```diff
@@ -41,6 +50,36 @@ func (c BaseConfig) GetTeleportConfig() lib.TeleportConfig {
 	return c.Teleport
 }
 
+func (c BaseConfig) GetTeleportClient(ctx context.Context) (teleport.Client, error) {
+	if validCred, err := credentials.CheckIfExpired(c.GetTeleportConfig().Credentials()); err != nil {
+		log.Warn(err)
+		if !validCred {
+			return nil, trace.BadParameter(
+				"No valid credentials found, this likely means credentials are expired. In this case, please sign new credentials and increase their TTL if needed.",
+			)
+		}
+		log.Info("At least one non-expired credential has been found, continuing startup")
+	}
+
+	bk := grpcbackoff.DefaultConfig
+	bk.MaxDelay = grpcBackoffMaxDelay
+
+	clt, err := client.New(ctx, client.Config{
+		Addrs:       c.GetTeleportConfig().GetAddrs(),
+		Credentials: c.GetTeleportConfig().Credentials(),
+		DialOpts: []grpc.DialOption{
+			grpc.WithConnectParams(grpc.ConnectParams{Backoff: bk, MinConnectTimeout: initTimeout}),
+			grpc.WithReturnConnectionError(),
+		},
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt = clt.WithCallOptions(grpc.WaitForReady(true))
```

Comment:
`WaitForReady` is now set globally for the entire client. Previously, it was set in some places:

* [Ping](https://github.com/gravitational/teleport-plugins/blob/931c006afe119289b17adcb3a8d564094f16df44/access/common/app.go#L96)
* [WatcherJob](https://github.com/gravitational/teleport-plugins/blob/a18e314d11457f2262ecff25e0621d4fbc955085/lib/watcherjob/watcherjob.go#L55)

But not in:

* [plugindata](https://github.com/gravitational/teleport-plugins/blob/a18e314d11457f2262ecff25e0621d4fbc955085/lib/plugindata/cas.go).

Not sure if this difference is intentional, please do point out if plugindata *requires* this option to be absent.

Reply:
I'm not aware of anything that could be different here